### PR TITLE
📧 Introduce new event - Gift Card Sent

### LINKED ISSUE #367: Implement `GIFT_CARD_SENT` async event
https://github.com/saleor/saleor/pull/12472

### PR DESCRIPTION
**Pull request by krzysztofwolski**
## Scope of the PR

Add support for sending Gift Card event.

Since it can be used with Saleor versions >=3.13, I've implemented a feature flag service, which returns a list of available features based on Saleor version.

When opening app with version <3.13, not available events are displayed as disabled, with additional message on hover: 
image

Besides that, the configuration is validated on save.

## Related issues

Closes #367

## Checklist

- [x] `.github/dependabot.yaml` is up-to date.
- [x] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/emails-and-messages/src/modules/smtp/default-templates.ts [hunk 1]
```diff
@@ -170,6 +170,22 @@ const defaultInvoiceSentMjmlTemplate = `<mjml>
   </mj-body>
 </mjml>`;
 
+// TODO: Improve the template
```
Comment (lkostrowski) [inline apps/emails-and-messages/src/modules/smtp/default-templates.ts:173]: its fine, users will do that

**Review by lkostrowski — COMMENTED** [review]
Please ensure this works with both newer and older Saleors

File: apps/emails-and-messages/package.json [hunk 2]
```diff
@@ -74,6 +74,6 @@
   },
   "private": true,
   "saleor": {
-    "schemaVersion": "3.11.7"
```
[the remark this answers is not in the document]
Comment (krzysztofwolski) [reply apps/emails-and-messages/package.json:77]: Thats a version of the GQL schema used for type generation. Added version check on register handler, with range 3.11.7 - 4.0.0. 
AFAIR 3.11.7 was related to webhooks - invoice and probably something else, don't recall exactly.

[the remark this answers is not in the document]
Comment (lkostrowski) [reply apps/emails-and-messages/package.json:77]: What I mean is that Saleor schema is not changing in patch versions

**Review by lkostrowski — APPROVED** [review]

**Pull request merged**